### DISCUSSION TITLE
how to get a BlockInfo by  fabric-gateway?

### DISCUSSION THREAD

**Discussion by diandiankou** [hyperledger/fabric-gateway#583] [Q&A]
i can get a BlockInfo through the fabric-gatway-Java repo, for example:

Channel channel = network.getChannel();
BlockInfo blockInfo = channel.queryBlockByNumber(Long.parseLong(txId));

But how can I get a BlockInfo through the fabric-gatway repo? I haven't seen any similar interfaces.

**Comment by bestbeforetoday** [comment]
You can use the fabric-gateway API to _evaluate_ the **GetBlockByNumber** transactions function on the **qscc** system chaincode. This returns a serialized [Block](https://hyperledger.github.io/fabric-protos/protos.html#common-Block) protocol buffer message. See the answer to [this question](https://github.com/hyperledger/fabric-gateway/discussions/504) for more details on how to unpack the Block.

An alternative is to use the [fabric-gateway API to retrieve blocks](https://hyperledger.github.io/fabric-gateway/main/api/java/org/hyperledger/fabric/client/Network.html) from the desired block number, read the first block, and then close the block iterator (unless you want to read subsequent blocks).
